### ISSUE TITLE
Switch to `pdoc` and update theme

### ISSUE THREAD

**Issue by davfsa** [hikari-py/hikari#808]
We should switch to [`pdoc`](https://github.com/mitmproxy/pdoc) with a new template to make the documentation easier to read and navigate. This should be done before #410 as the way pdoc outputs the HTML is a nicer and more compact format to allow for different version documentation.

Due to the huge scope of this, I will create a branch called `feature/pdoc` where progress will be made, which will later be merged into `master` once everything is ready.

### pdoc
- [x] Ensure everything renders correctly
  - `pdoc` uses a more strict numpy docstring style, so some things we are currently doing wont render nicecly anymore
    - `See also` section should only consist of links, no other text
- [x] Properties docstrings should not include a `Returns` section
- [x] Fix all the warnings printed (https://github.com/mitmproxy/pdoc/issues/307) [Python bug, so nothing we can do about it. Hopefully will be fixed in the future]
- [x] Ensure in-project linking works (relates to above)
- [x] `object.inv` output

### Theme
Still not decided. I have recently become a fan of rust projects documentation, as well as the nice usage of the sidebar in rustbook, so will probably take inspiration from there

#

*Both lists are not complete and can be edited to add more points as more issues are found or changes are to be made*

**Comment by FasterSpeeding** [comment]
I'ma just drop this script here cause this is a pretty nice way to get an easy to consume search json index on v8 pdoc https://gist.github.com/FasterSpeeding/19a6d3f44cdd0a1f3b2437a8c5eef07a (the result is the same as v7's search.json)

**Comment by davfsa** [comment]
Technically closed by #856, but will keep open till the branch is merged into master
